backend/server.py

```diff
@@ -49,7 +49,35 @@ def create(cls, hierarchy_data):
         db.session.add(new_hierarchy)
         db.session.commit() # hierarchy now has a unique id
 
-        return new_hierarchy.id
+        return new_hierarchy
+    
+    @classmethod
+    def get(cls, hierarchy_id):
+        return Hierarchy.query.filter_by(id=hierarchy_id).first()
+    
+    @classmethod
+    def get_dict(cls, hierarchy_id):
+        hierarchy = Hierarchy.get(hierarchy_id)
+        hier_dict = {
+            "id": str(hierarchy.id),
+            "name": hierarchy.name,
+            "description": hierarchy.description,
+        }
+
+        # Parse nodes and add list of dicts to hier_dict
+        hier_dict["nodes"] = Node.get_list(hierarchy_id, None)
+
+        return hier_dict
+    
+    @classmethod
+    def get_list(cls):
+        hierarchies = Hierarchy.query.all()
+        all_hierarchies = []
+
+        for hierarchy in hierarchies:
+            all_hierarchies.append(Hierarchy.get_dict(hierarchy.id))
+        
+        return all_hierarchies
 
 
 class Node(db.Model):
```

Comment:
For purposes of keeping things clean, I think we need to start splitting these into separate files. Maybe give each of these classes their own files if possible. I don't know how the sqlalchemy configuration wants it though.

Reply:
See below.